### PR TITLE
Ring signature support on Azimuth points

### PR DESCRIPTION
**Pull request by eglaysher**
This is the vere side of the ring signature support patch.

This mostly just using premade ECC math functions from ed25519...except for `jets/e/ge_additions.c`, which is an implementation of math primitives not provided by the ed25519. That implementation was translated from a Go implementation. I believe it's right, but please scrutinize it carefully.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -330,9 +335,27 @@ murmur3_dep = dependency('murmur3', version: '>=0.1.0', fallback: ['murmur3', 'm
 secp256k1_dep = dependency('libsecp256k1', version: '>=0.1.0', fallback: ['secp256k1', 'secp256k1_dep'])
 softfloat3_dep = dependency('softfloat3', version: '>=3.0.0', fallback: ['softfloat3', 'softfloat3_dep'])
 
+# Our ge additions must be in their own library because they cannot be compiled
+#  with '-funsigned-char' since they rely on signed characters.
```
Comment (joemfb) [inline meson.build]: On cc-release, this should probably be a separate derivation in `pkg/ge-additions` or somesuch.

**Comment by joemfb** [comment]
Extending my build comments in light of the CI requirements of urbit/arvo#1198:

To keep the urbit/arvo CI working, we need these changes as they are. To move the project forward, we want these changes on cc-release. The simplest thing is to both, but we could also selectively and temporarily disable some tests to navigate the interim period.

File: meson.build (hunk 1 above)
Comment (eglaysher) [reply meson.build]: Moved to cc-release and put this code in a separate `pkg/ge-additions` package.

File: jets/e/ed_add_double_scalarmult.c [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  u3qc_add_double_scalarmult(u3_atom a,
+                             u3_atom a_point,
+                             u3_atom b,
+                             u3_atom b_point)
+  {
+    c3_y met_w;
+
+    met_w = u3r_met(3, a);
+    if (met_w > 32) {
+      return u3m_bail(c3__exit);
```
[the remark this answers is not in the document]
Comment (eglaysher) [reply jets/e/ed_add_double_scalarmult.c]: If I add `?:  (gth (met 3 a) 32)  !!` lines to the hoon implementation, is `c3__exit` correct? I'm pretty sure even the hoon code won't work reliably on such large numbers, and there's already such `!!` checks in the `+ed` core (see `+puck` for an example.)

File: jets/e/ed_add_double_scalarmult.c [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    u3r_bytes(0, met_w, b_point_y, b_point);
+
+    ge_p3 A;
+    if (ge_frombytes_negate_vartime(&A, a_point_y) != 0) {
+      return u3m_bail(c3__exit);
+    }
+
+    ge_p3 B;
+    if (ge_frombytes_negate_vartime(&B, b_point_y) != 0) {
+      return u3m_bail(c3__exit);
```
[the remark this answers is not in the document]
Comment (eglaysher) [reply jets/e/ed_add_double_scalarmult.c]: I'm pretty sure it's correct to call `c3__exit` here. `ge_frombytes_negate_vartime` (plus the `fe_neg` calls below) correspond to the `(need (deco ...))` lines in the hoon implementation.

File: jets/e/ed_add_double_scalarmult.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (eglaysher) [reply jets/e/ed_add_double_scalarmult.c]: Actually, skimming this again, I'm not confident this is right. I'm going to `__fail` for now.

**Comment by eglaysher** [comment]
PTAL. This is now based on cc-release and I went with changing the exits to fails in the case of the `u3r_met()`s after all.

**Pull request merged**